testbot.py

```diff
@@ -160,4 +168,56 @@ async def join_game(arguments: list, message: discord.message):
     logging.info(f'Assigned {message.author} to {game_name}')
     await message.channel.send(f'Have fun playing {game_name}, {message.author}!')
 
+
+async def format_poll(message: discord.message):
+    command = message.content.replace("pb poll", '').strip()
+
+    if "\"" not in command or ',' not in command:
+        logging.debug('No poll title or arguments were passed')
+        await message.channel.send("You can't just make a poll without a title or arguments! Please insert a poll "
+                                   "title that is surrounded in \"\" as well as arguments proceeded with commas.")
+        return
+
+    title = command[command.find("\"") + 1:command.rfind("\"")]
+    options = command.replace(f'\"{title}\"', '').strip().split(", ")
+    logging.info(command)
+    poll = {
+        "title": title,
+        "options": options
+    }
+    logging.info('title --' + poll.get("title"))
+    logging.info(poll.get("options"))
+    return poll
+
+
+async def make_poll(arguments: dict, message: discord.message):
+    if not arguments:
+        logging.debug('No arguments passed')
+        await message.channel.send("A poll has to have a title and options to vote on!")
+        return
+
+    options = arguments.get("options")
+    argument_length = len(options)
+
+    if argument_length > 8:
+        logging.debug('Too many arguments passed')
+        await message.channel.send("Too many poll options! The maximum amount of poll options you can add is 8!")
+        return
+
+    poll_title = arguments.get("title")
+    color = discord.Colour(random.randint(0, 0xFFFFFF))
+    emojis = ['🔴', '🟡', '🟢', '🔵', '🟣', '🟤', '⚫', '⚪']
+    option_string = ''
+    for i in range(argument_length):
+        options[i] = f"{emojis[i]} {options[i]} \n"
+        option_string = f'{option_string} {options[i]}'
+
+    poll_embed = discord.Embed(title=poll_title, description=option_string, color=color)
+    logging.info(f'Sending message')
+    bot_poll = await message.channel.send(embed=poll_embed)  # A bot sending a message actually returns the message!
+
+    for i in range(argument_length):
+        await bot_poll.add_reaction(emojis[i])
```

Comment:
Not sure if python has the same ability, but avoid using await in loops. `await` allows for asynchronous programming, but in this style of awaiting for every iteration, it has become synchronous. 

Further reading: https://eslint.org/docs/rules/no-await-in-loop

Reply:
Check this method https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

You can combine all these calls into an array first, then use the asyncio.gather method to start them all concurrently

Reply:
We just discussed. Not needed to make it asynchronous because we want the reactions to be added sequentially (synchronous style) rather than the emojis to be added in a random order.

Reply:
Good catch!